__tests__/statisticsRoutes.test.js

```diff
@@ -0,0 +1,186 @@
+import sinon from 'sinon';
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '..';
+import models from '../database/models';
+
+chai.use(chaiHttp);
+
+const url = '/api/v1';
+const { expect } = chai;
+const { Article, Reading } = models;
+const blacklistedToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZmlyc3ROYW1lIjoiSm9obiIsImlhdCI6MTU2NDAwOTA5NCwiZXhwIjoxNTY0MDEyNjk0fQ.J5ktoXlmLxOtV8R16sNPMXXeydwRdCA8h6Cep-AzZnc';
+
+
+describe('Reading statistics for articles published by an author', () => {
+  let userToken;
+  before((done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signin')
+      .send({ email: 'john.kamali@outlook.com', password: 'P455w0rd' })
+      .end((err, res) => {
+        userToken = res.body.user.token;
+        done();
+      });
+  });
+  it('should return an array of objects in another object', (done) => {
+    chai.request(app)
+      .get(`${url}/statistics/published`)
+      .set('Authorization', `Bearer ${userToken}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(200);
+        expect(res.body.allArticlesStatistics[1].id).to.eql('a2cda7e8-28ba-4507-9563-3e4ea280efb6');
+        expect(res.body.allArticlesStatistics[1].title).to.eql('Skull is coming to town');
+        expect(res.body.allArticlesStatistics[1].slug).to.eql('Skull-is-9563-3e4ea280efb6');
+        expect(res.body.allArticlesStatistics[1].userId).to.eql('e89695aa-1f7e-468c-89dd-5fc2332e31f1');
+        expect(res.body.allArticlesStatistics[1].readCount).to.eql(0);
+        done();
+      });
+  });
+  it('should return an error response when token is not provided', (done) => {
+    chai.request(app)
+      .get(`${url}/statistics/published`)
+      .set('Authorization', '')
+      .end((err, res) => {
+        expect(res.status).to.eql(401);
+        expect(res.body.errors.message).to.eql('No token provided');
+        done();
+      });
+  });
+  it('should return an error response when an invalid token is provided', (done) => {
+    chai.request(app)
+      .get(`${url}/statistics/published`)
+      .set('Authorization', `Bearer ${userToken}s`)
+      .end((err, res) => {
+        expect(res.status).to.eql(403);
```

Comment:
and using 403 here is there a difference? you should change them to use a single error code. every other thing LGTM

Reply:
@VictorMaria see this 👆🏽

You still have some `403`s returned for invalid and blacklisted tokens in your test cases